pac4j-core/src/main/java/org/pac4j/core/authorization/generator/FromAttributesAuthorizationGenerator.java

```diff
@@ -1,72 +1,80 @@
 package org.pac4j.core.authorization.generator;
 
+import org.pac4j.core.profile.CommonProfile;
+
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.StringTokenizer;
 
-import org.pac4j.core.profile.CommonProfile;
-
 /**
  * <p>Generate the authorization information by inspecting attributes.</p>
  * <p>The attributes containing the roles separated by the {@link #splitChar} property (can be set through {@link #setSplitChar(String)}) are
  * defined in the constructor. It's the same for the attributes containing the permissions.</p>
- * 
+ *
  * @author Jerome Leleu
  * @since 1.5.0
  */
 public class FromAttributesAuthorizationGenerator<U extends CommonProfile> implements AuthorizationGenerator<U> {
-    
-    private final String[] roleAttributes;
-    
-    private final String[] permissionAttributes;
-    
+
+    private final Iterable<String> roleAttributes;
+
+    private final Iterable<String> permissionAttributes;
+
     private String splitChar = ",";
-    
-    public FromAttributesAuthorizationGenerator(final String[] roleAttributes, final String[] permissionAttributes) {
-        this.roleAttributes = copyArray(roleAttributes);
-        this.permissionAttributes = copyArray(permissionAttributes);
+
+    public FromAttributesAuthorizationGenerator(final Iterable<String> roleAttributes, final Iterable<String> permissionAttributes) {
+        this.roleAttributes = roleAttributes;
+        this.permissionAttributes = permissionAttributes;
     }
 
-    private String[] copyArray(final String[] original) {
-        if (original != null) {
-            return Arrays.copyOf(original, original.length);
+    public FromAttributesAuthorizationGenerator(final String[] roleAttributes, final String[] permissionAttributes) {
+        if (roleAttributes != null) {
+            this.roleAttributes = Arrays.asList(roleAttributes);
```

Comment:
Careful, the behaviour is a little changed: `Arrays.copyOf` was making a copy, while `Arrays.asList` use the array itself as the backend of an ArrayList; changes to the array will be reflected in the `ArrayList`.